### PR TITLE
[3.0] Various deprecations cleanups

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -901,10 +901,7 @@ private function relativize($xpath)
                 $expression = substr($expression, strlen($parenthesis));
             }
 
-            // BC for Symfony 2.4 and lower were elements were adding in a fake _root parent
-            if (0 === strpos($expression, '/_root/')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DomCrawler/Crawler.php]: See #16109

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 2]
```diff
@@ -217,7 +217,7 @@ private function parseDefinition(\DOMElement $service, $file)
                 if (false !== strpos($name, '-') && false === strpos($name, '_') && !array_key_exists($normalizedName = str_replace('-', '_', $name), $parameters)) {
                     $parameters[$normalizedName] = XmlUtils::phpize($node->nodeValue);
                 }
-                // keep not normalized key for BC too
+                // keep not normalized key
                 $parameters[$name] = XmlUtils::phpize($node->nodeValue);
```
Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: shouldn't this be removed?

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: I'm not sure it's worth the burden for users. What do others think?

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**